### DISCUSSION TITLE
Problematic class which seems to be breaking everything

### DISCUSSION THREAD

**Discussion by thebxzr** [AssetRipper/AssetRipper#1238] [Q&A]
Hi, I've been trying to decompile a Unity game, and am almost finished, however I've stumbled upon a weird dummy class.
```cs
using UnityEngine;

namespace Rewired.Components
{
	public class ComponentWrapper`1 : MonoBehaviour
	{
		/*
		insert text - a bunch of problems and solutions
		*/
	}
}
```

One thing I've noticed is that there's a backquote and one in its name, which should not normally be possible or allowed.

For reference, I opened it in dnSpy and found this:
![image](https://github.com/AssetRipper/AssetRipper/assets/67138856/0b31deb7-7976-4b38-b654-7eab49e2ee9e)

Other pieces of code rely on this too, so I can't delete it.
Every single other piece of code seems to have exported fine.
My hunch is that it's a file meant to throw people off and make it harder to decompile. If anyone can help me, that would be greatly appreciated!

**Comment by ds5678** [comment]
Normally, that means a class is generic, ie `ComponentWrapper<T>`.

**Reply by thebxzr** [reply]
how exactly would i fix it?

**Reply by ds5678** [reply]
If it's actually not generic, then you just remove that part.

**Reply by thebxzr** [reply]
1) How do I know if it's generic?
2) When I remove it, ~130 errors appear

**Reply by ds5678** [reply]
1. It probably is not generic if ilspy says it's not.
2. Then you have fix those too.

**Reply by thebxzr** [reply]
the problem is that those errors look like garbage, with variables name that are just a random mess of characters, i'll send some later

**Reply by ds5678** [reply]
I don't understand why you're in this situation in the first place. Hybrid export is ideal for obfuscated assemblies like Rewired.

**Reply by thebxzr** [reply]
I'm trying to open it in Unity -- somebody told me Decompiled was better than Hybrid

**Reply by ds5678** [reply]
It is not

**Reply by thebxzr** [reply]
should i use hybrid?

**Reply by thebxzr** [reply]
well i tried hybrid and I got the exact same result, only with more errors and it crashes whenever I open it or try adding a package like Unity UI

**Reply by thebxzr** [reply]
for reference, i tried opening it in not safe mode because that worked for decompile, but it crashed. I opened it in safe mode, and added Unity UI and it crashes whenever I open it

**Reply by thebxzr** [reply]
okay I got it into unity and put all the dll's into the assets folder, but the problem is some scripts are randomly missing. also the new comment I just made about the problem, too

**Comment by dukemagus** [comment]
Rewired is a pretty popular unity asset. You can buy the original on the store and replace the offending scripts.

**Reply by thebxzr** [reply]
that's a good idea but the problem is i'm really not ready to splash out 50+ dollars just to decompile a game. thank for the help though!

**Comment by thebxzr** [comment]
I made some progress, but now I have a problem where it says "Extension already registered for type, Parameter name: type". This occurs for UnityEngine.GUISkin and UnityEngine.Playables.PlayableAsset. I looked it up and the only thing I found was an old thread from 2021.

**Reply by thebxzr** [reply]
Well I messed around with libraries, and now the game is (somewhat) functional. All components have had their values (in most cases) be None'd. Is there anything I might've did wrong, because I would like to get the correct values back

**Reply by thebxzr** [reply]
also the graphics are weird, ui elements are weirdly colored, and text is blocky instead of having proper graphics

**Reply by ds5678** [reply]
That sounds like an issue with not using the package

**Reply by thebxzr** [reply]
which package? the game uses a lot
i went into AuxilaryFiles and added the dll's into Assets, then replaced some with their Package Manager counterparts

**Reply by thebxzr** [reply]
^ i replaced some with their Package Manager ones because, i.e. TMPro didn't recognise things at TMPro
